src/components/BreedsList.vue

```diff
@@ -93,69 +94,183 @@ export default {
+      });
+      this.displayedBreeds.forEach((breed) => {
+        breed.isActive = false;
+      });
+      if (this.$route.name !== "breeds") {
+        this.$router.push({
+          name: "breeds",
+        });
+      }
+    },
+    toggleCard(breed, refLi) {
+      breed.isActive = !breed.isActive;
+      refLi.scrollIntoView({
+        behavior: "smooth",
+        block: "center",
+      });
+      this.displayedBreeds.forEach((item) => {
+        if (item.id !== breed.id) {
+          item.isActive = false;
+        }
+      });
+    },
+    showBreedDetailsOnRouteParam(breedName) {
+      if (breedName && this.$refs.resultListContainer) {
+        this.displayedBreeds.forEach((breed, id) => {
+          if (breed.name.toLowerCase() === breedName.toLowerCase()) {
+            breed.isActive = true;
+            this.$nextTick(() => {
+              this.$refs.resultListContainer.children[
+                id
+              ].firstChild.scrollIntoView({
+                behavior: "smooth",
+                block: "center",
+              });
+            });
+          } else {
+            breed.isActive = false;
+          }
+        });
+      }
     },
   },
   computed: {
     displayedBreeds() {
-      if (this.searchList.length > 0) {
-        return this.searchList;
-      } else if (this.searchList.length === 0 && this.searchQuery !== "") {
-        this.searchList = [];
+      if (this.searchQuery) {
         return this.searchList;
       } else {
         return this.breedsList;
       }
     },
+    loadingState() {
+      if (this.isLoading) {
+        return this.message;
+      } else if (!this.isLoading && this.error) {
+        return this.error;
+      }
+    },
+    receivedDataState() {
+      if (!this.displayedBreeds.length && !this.isLoading) {
+        return this.message;
+      }
+    },
+  },
+  watch: {
+    breedName(newName, oldName) {
+      if (newName !== oldName) {
+        this.showBreedDetailsOnRouteParam(newName);
+      }
+    },
+
+    "$route.query.q": {
+      immediate: true,
+      handler(newName, oldName) {
+        if (newName && newName !== oldName) {
+          this.searchQuery = newName;
+          this.getSearchResults(this.searchQuery);
+        }
+      },
+    },
   },
 
   created() {
     this.fetchData();
   },
+  mounted() {
+    if (this.$route.name === "search") {
+      this.getSearchResults(this.searchQuery);
+    }
+  },
+
+  updated() {
+    if (!this.isScrollToTopActive && this.$route.name === "breedName") {
+      this.showBreedDetailsOnRouteParam(this.breedName);
+      this.searchQuery = "";
+    }
+  },
+
+  beforeRouteLeave(to, from, next) {
+    if (from.name === "search") {
+      this.searchQuery = "";
+
+      this.$refs.searchRef.$el[0].value = "";
+      next();
+    }
+    if (from.name === "breedName") {
+      this.displayedBreeds.forEach((breed) => {
+        breed.isActive = false;
+      });
+      next();
+    }
+
+    next();
+  },
```

Comment:
Hotel sem označiti več stvari, pa ni šlo ;) 
Naredil sem:
- Implementiral sem /breed route, torej, ko uporabnik odpre app, je na /breeds route-u, izpišejo se mu vse pasme
- ko uporabnik klikne na card, se mu odpre ta card, route je /breed/:breedName, torej je dinamičen route parameter, torej odvisno od pasme ki jo klikne, npr. http://127.0.0.1:5173/breeds/Akita, pasma ki jo klikne se scrolla na sredino
-ko uporabnik išče po pasmah, mu izpiše zadetke in route se spremeni v /search?q=aki, torej se spreminja glede na search input, če je samoen zadetek mu le tega tudi odpre
- če gre uporabnik nazaj oz naprej z gumb v browserju ga postavi na prejšenji routes oz naslednji in tudi scrolla na izbran card
- če uporabnik ročno vnese route, npr /breeds/akita, ga vrze na to pasmo in scrolla na sredino, tudi če uporabni ročno vnese
npr. /search?q=aki, mu izpiše vse zadetke
Če uporabnik klikne gumb, scrollToTop, da ga vrže na začetek ul, se route spremeni na /breeds in vsi cardsi so zaprti